wolfSSL/src/chapter02.md

```diff
@@ -528,18 +696,188 @@ Enables Elliptical Curve Cryptography (ECC) support.
 
+#### HAVE_EXT_CACHE
+
+Enables a feature support use of an external session cache (vs an internal one).
+
+#### WOLFSSL_WPAS_SMALL
+
+Enables a smaller subset of the compatibility layer for WPA supplicant support.
+
+#### OPENSSL_ALL
+
+Enables support for all compatibility functions for testing integration.
+
+#### OPENSSL_COEXIST
+
+OpenSSL compat layer. Needs old names disabled. Mode to allow wolfSSL and OpenSSL to exist together.
+
+#### OPENSSL_VERSION_NUMBER
+
+Specifies the version number to implement OpenSSL compatibility.
+
+#### WOLFSSL_NGINX
+
+OpenSSL compatibility application specific. Use, nginx `(--enable-nginx) WOLFSSL_NGINX`.
+
+#### WOLFSSL_ERROR_CODE_OPENSSL
+
+Open SSL compatibility API wolfSSL_EVP_PKEY_cmp returns 0 on success and -1 on failure. This behavior is different from OpenSSL. EVP_PKEY_cmp returns:
+1: two keys match
+0: do not match
+-1: key types are different
+-2: the operation is not supported
+If you want this function to behave the same as openSSL, define WOLFSSL_ERROR_CODE_OPENSSL so that WS_RETURN_CODE translates return codes to match OpenSSL equivalent behavior.
+
+#### WOLFSSL_HARDEN_TLS
+
+Implement the recommendations specified in RFC9325. This macro needs to be defined to the desired number of bits of security. The currently implemented values are 112 and 128 bits. The following macros disable certain checks.
+- WOLFSSL_HARDEN_TLS_ALLOW_TRUNCATED_HMAC
+- WOLFSSL_HARDEN_TLS_ALLOW_OLD_TLS
+- WOLFSSL_HARDEN_TLS_NO_SCR_CHECK
+- WOLFSSL_HARDEN_TLS_NO_PKEY_CHECK
+- WOLFSSL_HARDEN_TLS_ALLOW_ALL_CIPHERSUITES
+
+#### WOLFSSL_ASIO
+
+OpenSSL compatibility specific macro.
+
+#### WOLFSSL_QT
+
+OpenSSL compatibility specific. Enable DH Extra for QT, OpenSSL all, OpenSSH, and static ephemeral.
+
+#### WOLFSSL_HAPROXY
+
+OpenSSL compatibility specific macro.
+
+#### WOLFSSL_ASN_TEMPLATE
+
+Use newer ASN template asn.c code (default). Daul algo certificate features. Dual alg cert support requires the ASN.1 template feature.
+
+#### WOLFSSL_ASYNC_IO
+
+Used in async cleanup.
+
+#### WOLFSSL_ATMEL
+
+Enables ASF hooks seeding random data using the `atmel_get_random_number` function.
+
+#### WOLFSSL_CMAC
+
+Additional CMAC algorithm enable. Note: requires `WOLFSSL_AES_DIRECT`.
+
+#### WOLFSSL_ESPIDF_ERROR_PAUSE
+
+Enable to pause in a loop rather than exit.
+
 #### TEST_IPV6
 
 Turns on testing of IPv6 in the test applications. wolfSSL proper is IP neutral, but the testing applications use IPv4 by default.
 
+#### TEST_NONBLOCK_CERTS
+
+Async test --enable-ocsp CFLAGS implement `TEST_NONBLOCK_CERTS`.
+
+#### TEST_OPENSSL_COEXIST
+
+Use when enabling the build option: `./configure --enable-opensslcoexist`.
+
+#### TEST_PK_PRIVKEY
+
+Loads and test PK private key.
+
+#### TEST_BUFFER_SIZE
+
+Can allocate buffer size for benchmarking.
+
+#### FORCE_BUFFER_TEST
+
+"Forces" buffer test in benchmark.
```

Comment:
Forces use of the test_certs.h buffers instead of using the file system. Used for internal testing only in wolfssl/test.h.